internal/logic/logic_test.go

```diff
@@ -51,73 +61,257 @@ func Test_transactionManagementServiceLogic_HealthCheck(t *testing.T) {
 	}
 }
 
-func Test_transactionManagementServiceLogic_Ping(t *testing.T) {
+func TestTransactionManagementServiceLogic_GetTransactions(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
 
 	tests := []struct {
-		name  string
-		setup func() datasource.DataSource
-		give  *model.PingRequest
-		want  *respModel.Response
+		name   string
+		userId string
+		setup  func() (datasource.DataSourceI, string)
+		want   func(*respModel.Response)
 	}{
 		{
-			name: "Success",
-			setup: func() datasource.DataSource {
-				mockDs := mock.NewMockDataSource(mockCtrl)
+			name:   "Success :: Get Transaction",
+			userId: "123",
+			setup: func() (datasource.DataSourceI, string) {
+				mockDs := mock.NewMockDataSourceI(mockCtrl)
+				var trans []model.Transaction
+				trans = append(trans, model.Transaction{UserId: "123", AccountNumber: 1})
+				mockDs.EXPECT().Get(map[string]interface{}{"user_id": "123"}, 5, 0).Times(1).Return(trans, 1, nil)
+				return mockDs, ""
+			},
+			want: func(resp *respModel.Response) {
+				var users = model.GetTransaction{AccountNumber: 1}
+				temp := respModel.Response{
+					Status:  http.StatusOK,
+					Message: "SUCCESS",
+					Data:    users,
+				}
+				res, ok := resp.Data.(model.PaginatedResponse)
```

Comment:
Compare the string data as a whole in the form of string instead of comparing every field one-by-one.
Reflect across the other test cases.

Reply:
done

Reply:
I used a separate struct and compared all fields at once